src/worker/start.rs

```diff
@@ -372,30 +370,29 @@ fn try_get_slurm_info() -> anyhow::Result<Map<String, String>> {
             anyhow!("SLURM_JOB_ID/SLURM_JOBID not found. The process is not running under SLURM")
         })?;
 
-    let mut result = Map::with_capacity(2);
-    result.insert("MANAGER".to_string(), "SLURM".to_string());
-    result.insert("MANAGER_JOB_ID".to_string(), manager_job_id);
+    log::info!("SLURM environment detected");
 
     // TODO: Get walltime info
-
-    log::info!("SLURM environment detected");
-    Ok(result)
+    let duration = Duration::from_secs(1);
+    Ok(ManagerInfo::new(
+        ManagerType::Slurm,
+        manager_job_id,
+        duration,
+    ))
 }
 
-fn gather_manager_info(opts: ManagerOpts) -> anyhow::Result<Map<String, String>> {
+fn gather_manager_info(opts: ManagerOpts) -> Option<ManagerInfo> {
     match opts {
         ManagerOpts::Detect => {
             log::debug!("Trying to detect manager");
-            try_get_pbs_info()
-                .or_else(|_| try_get_slurm_info())
-                .or_else(|_| Ok(Map::new()))
+            try_get_pbs_info().or_else(|_| try_get_slurm_info()).ok()
         }
         ManagerOpts::None => {
             log::debug!("Manager detection disabled");
-            Ok(Map::new())
+            None
         }
-        ManagerOpts::Pbs => try_get_pbs_info(),
-        ManagerOpts::Slurm => try_get_slurm_info(),
+        ManagerOpts::Pbs => Some(try_get_pbs_info().expect("Failed to get PBS info")),
```

Comment:
No panic please

Reply:
But the program should still crash, right?

Reply:
It should terminated with an error.